### PR TITLE
Connecting the frontend with backend for register and login

### PR DESCRIPTION
**Pull request by Martincho2003**
I have made slight correction in you js files(ex. login and register send POST method) and the links for the request. You have to make other changes that you can see in the issues.

### CODE REVIEW

File: frontend/src/components/Login.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ class Login extends Component {
 	  }
 
 	  checkUser = async () => {
-		let res = await api.get('/', { email: this.state.email, password: this.state.password})
+		let res = await api.post('/', { email: this.state.email, password: this.state.password})
```
Comment (marto55) [inline frontend/src/components/Login.js:43]: According to REST and the CRUD standart, you should use GET to 'Read' information and POST to 'Create' stuff. Here you don't make new entry in the database, you only get answer which counts as reading. 

So it shoud be GET

Comment (Martincho2003) [reply frontend/src/components/Login.js:43]: U have a point but i read in some forums that mostof the people use POST for login and register

Comment (Martincho2003) [reply frontend/src/components/Login.js:43]: I am not sure why but my logic is that, usually i make a GET request to load the login/register page and then the submit button sends POST request for the login/register

Comment (marto55) [reply frontend/src/components/Login.js:43]: Fine it will be your way

File: chats/views.py [hunk 2]
```diff
@@ -39,55 +43,79 @@ def index(request):
...
+                                    "is_active": current_user.is_active})
         else:
             print("not logged")
-            return render(request, "login.html")
-    else:
-       return render(request, "login.html") 
+            return JsonResponse({"status" : "unsuccessfull"})
 
 def log_out(request):
     logout(request)
```
Comment (marto55) [inline chats/views.py:121]: Please comment the way your body looks. Whaen I try the /login request with:
{
    "username": "Marto",
    "password": "12345678"
}
It returns:
{
    "status": "unsuccessfull"
}
So I guess the reques might have to be different.

File: chats/views.py [hunk 3]
```diff
@@ -39,55 +43,79 @@ def index(request):
...
         myusr.first_name = fname
         myusr.last_name = lname
         myusr.is_active = True
         myusr.save()
-        messages.success(request, "Your account has been created succesfully!")
-        return redirect("login")
-
-    return render(request, "register.html") 
+        #messages.success(request, "Your account has been created succesfully!")
+        return JsonResponse({"status" : "successfull"})
```
Comment (marto55) [inline chats/views.py:95]: Please comment the way your body looks. Whaen I try the /login request with:
{
    "username": "Martinakis",
    "first_name": "m1",
    "last_name": "m2",
    "email": "m@m",
    "password1": "m3",
    "password2": "m3"
}
It returns error page and code 500 "Internal server error"
It also loggs:
django.utils.datastructures.MultiValueDictKeyError: 'username'
[12/Feb/2022 15:42:53] "POST /chats/register HTTP/1.1" 500 72636

So I guess the reques might have to be different.

File: chats/views.py (hunk 2 above)
Comment (marto55) [reply chats/views.py:121]: "Marto" with password "12345678" is existing user. The server loggs correct received username and password.

File: chats/views.py (hunk 3 above)
Comment (marto55) [reply chats/views.py:95]: goood its fixed

File: chats/views.py (hunk 2 above)
Comment (marto55) [reply chats/views.py:121]: good we fixed it

**Review by marto55 — APPROVED** [review]

**Pull request merged**